kong/api/routes/certificates.lua

```diff
@@ -1,317 +1,80 @@
-      end
-    end
-
-    ssl_cert.snis[i] = sni
-  end
 
-  -- delete snis which should no longer use ssl_cert
-  for i = 1, #old_snis do
-    if not snis_in_request[old_snis[i].name] then
-      -- ignoring error
-      -- if we want to return an error here
-      -- to return 4xx here, the current transaction needs to be
-      -- rolled back else we risk an invalid state and confusing
-      -- the user
-      dao_factory.ssl_servers_names:delete({
-        name = old_snis[i].name,
-      })
-    end
+    self.params.certificates = cert.id
   end
-
-  return helpers.responses.send_HTTP_OK(ssl_cert)
 end
 
 
 return {
-  ["/certificates/"] = {
-    POST = function(self, dao_factory, helpers)
-      create_certificate(self, dao_factory, helpers)
+  ["/certificates"] = {
+    -- override to include the snis list when getting all certificates
+    GET = function(self, db, helpers)
+      local data, _, err_t, offset =
+        db.certificates:page_with_name_list(self.args.size,
+                                            self.args.offset)
+      if not data then
+        return endpoints.handle_error(err_t)
+      end
+
+      local next_page = offset and string.format("/certificates?offset=%s",
+                                                 ngx.escape_uri(offset)) or ngx.null
+
+      return helpers.responses.send_HTTP_OK {
+        data   = data,
+        offset = offset,
+        next   = next_page,
+      }
     end,
 
-
-    GET = function(self, dao_factory, helpers)
-      local ssl_certificates, err = dao_factory.ssl_certificates:find_all()
-      if err then
-        return helpers.yield_error(err)
+    -- override to accept the snis param when creating a certificate
+    POST = function(self, db, helpers)
+      local data, _, err_t = db.certificates:insert_with_name_list(self.args.post)
+      if err_t then
+        return endpoints.handle_error(err_t)
       end
 
-      for i = 1, #ssl_certificates do
-        local rows, err = dao_factory.ssl_servers_names:find_all {
-          ssl_certificate_id = ssl_certificates[i].id
-        }
-        if err then
-          return helpers.yield_error(err)
-        end
-
-        ssl_certificates[i].snis = setmetatable({}, cjson.empty_array_mt)
-
-        for j = 1, #rows do
-          ssl_certificates[i].snis[j] = rows[j].name
-        end
-      end
-
-      return helpers.responses.send_HTTP_OK({
-        data = #ssl_certificates > 0 and ssl_certificates or cjson.empty_array,
-        total = #ssl_certificates,
-      })
-    end,
-
-
-    PUT = function(self, dao_factory, helpers)
-      -- no id present, behaviour should be same as POST
-      if not self.params.id then
-        create_certificate(self, dao_factory, helpers)
-        return -- avoid tail call
-      end
-
-      -- id present in body
-      update_certificate(self, dao_factory, helpers)
+      return helpers.responses.send_HTTP_CREATED(data)
     end,
   },
 
+  ["/certificates/:certificates"] = {
```

Comment:
I think you need to use what @kikito used to override autogenerated.

Reply:
I confirm that is the case. The auto-generated route is `"certificates/certificates"` so I must use the same name in order to override stuff.

Reply:
That is exactly the reason. The auto-generated route is `/certificates/:certificates`.

We would have had to add a param called "singular_name" or something similar to the schema in order to be able to do fancy things like sometimes using singulars instead of plurals everywhere.